### ISSUE TITLE
Double spending transaction - brs.BlockchainProcessorImpl - Block not accepted

### ISSUE THREAD

**Issue by matbogdan79** [PoC-Consortium/burstcoin#382]
Every couple of days I get this on some of my peers:
[INFO] 2018-08-22 09:43:35 brs.peer.PeerImpl - Blacklisting 80.184.140.195 (2.2.2) because of: found invalid pull/push data during importing the block
[SEVERE] 2018-08-22 09:44:03 brs.BlockchainProcessorImpl - Block not accepted
brs.BlockchainProcessor$TransactionNotAcceptedException: Double spending transaction: 1610611383476210119 transaction: {"senderPublicKey":"21b1c5aac5daf68aad0a8753165bb68169ac3bb41bde2ffb87d8404711519f6f","signature":"d477bd64a221ad77f63222a93f8b535d070a5eddeffc9442dac2769b6b6781037e30631fddd2021ddef68f7832893372a1ab40ae9beea42f6eb8310fc8742218","feeNQT":10000000,"type":0,"version":1,"ecBlockId":"2278637779915525718","attachment":{},"subtype":0,"amountNQT":84823500000,"recipient":"243989817010793960","ecBlockHeight":525800,"deadline":1440,"timestamp":127181847}
        at brs.BlockchainProcessorImpl.accept(BlockchainProcessorImpl.java:1017)
        at brs.BlockchainProcessorImpl.pushBlock(BlockchainProcessorImpl.java:993)
        at brs.BlockchainProcessorImpl.lambda$new$8(BlockchainProcessorImpl.java:290)
        at brs.util.ThreadPool.lambda$start$0(ThreadPool.java:96)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[INFO] 2018-08-22 09:44:03 brs.peer.PeerImpl - Blacklisting 206.54.172.55 (2.2.2) because of: found invalid pull/push data during importing the block

Sometimes a simple wallet restart fix it, sometimes I have to do deeper popoff and sometimes I have to cleanDB and re-import.

Lucky me that the pools have more than 1-2 peers and the mining operation is not affected but this need to be fixed because the wallet is getting stuck if this error comes.
Basically it bans all the peers it has one by one.

More than this , the wallet remain on block 525806 and keep complaining about transaction 1610611383476210119

**Comment by nixops** [comment]
What platform?

**Comment by Brabantian** [comment]
Should be fixed due to better synchronization of Threads
